beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -145,6 +184,27 @@ impl<T: BeaconChainTypes> DataAvailabilityChecker<T> {
             .put_kzg_verified_blobs(block_root, verified_blobs)
     }
 
+    /// Put a list of custody columns received via RPC into the availability cache. This performs KZG
+    /// verification on the blobs in the list.
+    pub fn put_rpc_custody_columns(
+        &self,
+        block_root: Hash256,
+        custody_columns: Vec<CustodyDataColumn<T::EthSpec>>,
+    ) -> Result<Availability<T::EthSpec>, AvailabilityCheckError> {
+        let Some(kzg) = self.kzg.as_ref() else {
+            return Err(AvailabilityCheckError::KzgNotInitialized);
+        };
+
+        // TODO(das): report which column is invalid for proper peer scoring
+        let verified_custody_columns = custody_columns
```

Comment:
we could do batch KZG verification here

Reply:
ah, we can't really with the current construct, as they are done in the individual `KzgVerifiedDataColumn` constructors. We'd probably need an extra struct to hold the list if we need want this.